homeassistant/components/switcher_kis/sensor.py

```diff
@@ -0,0 +1,251 @@
+        sensors_for_register = {}  # type: Dict[str, Entity]
+
+        for i in range(8):
+            idx = str(i)
+            entity_created = SwitcherScheduleSensor(
+                hass, SCHEDULE_NAME_FORMAT.format(name, idx),
+                SCHEDULE_FRIENDLY_NAME_FORMAT.format(friendly_name, idx),
+                icon, device.ip_addr, device.phone_id, device.device_id,
+                device.device_password, idx)
+
+            if discovery_info.get(DISCOVERY_SCHEDULES):
+                for schedule in discovery_info[DISCOVERY_SCHEDULES]:
+                    if schedule.schedule_id == idx:
+                        await entity_created.async_update_schedule(
+                            schedule, True)
+                        break
+
+            sensors_created.append(entity_created)
+            sensors_for_register[idx] = entity_created
+
+    else:
+        raise PlatformNotReady("No device data discoverd")
+
+    async_add_entities(sensors_created, False)
+
+    try:
+        await wait_for(async_register_sensor_entities(sensors_for_register),
+                       timeout=1)
+
+    except AsyncioTimeoutError:
+        raise PlatformNotReady("Unable to register for data updates")
+
+    return None
+
+
+class SwitcherScheduleSensor(Entity):
+    """Representation of the schedule sensor entity."""
+
+    def __init__(self, hass: HomeAssistant, name: str, friendly_name: str,
+                 icon: str, ip_addr: str, phone_id: str, device_id: str,
+                 device_password: str, schedule_id: str) -> None:
+        """Initialize the sensor."""
+        self._hass = hass
+        self._ip_addr = ip_addr
+        self._phone_id = phone_id
+        self._device_id = device_id
+        self._device_password = device_password
+        self._schedule_id = schedule_id
+        self.entity_id = async_generate_entity_id(
+            ENTITY_ID_FORMAT, name, hass=hass)
+        self._name = friendly_name
+        self._icon = icon
+        self._configured = False
+        self._schedule_details = None  # type: Optional[SwitcherV2Schedule]
+        self._next_run = STATE_NOT_RUNNING
+
+    @property
+    def name(self) -> str:
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def state(self) -> str:
+        """Return the state of the sensor."""
+        if not self._configured:
+            return STATE_NOT_CONFIGURED
+        if self._schedule_details and not self._schedule_details.enabled:
+            return STATE_NOT_ENABLED
+        return self._next_run
+
+    @property
+    def should_poll(self) -> bool:
+        """No polling needed."""
+        return False
+
+    @property
+    def icon(self) -> str:
+        """Return mdi icon."""
+        return self._icon
+
+    @property
+    def schedule_id(self) -> str:
+        """Return the schedule id."""
+        return self._schedule_id
+
+    @property
+    def state_attributes(self) -> Optional[Dict]:
```

Comment:
Don't override this property.  Return your custom items in `device_state_attributes`.